flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/ExecutingTest.java

```diff
@@ -155,11 +155,53 @@ public void testTransitionToFinishedOnFailedExecutionGraph() throws Exception {
         }
     }
 
+    @Test
+    public void testEnsureOnlyGlobalTerminalStateCallbacksNoCall() throws Exception {
+        // modify MockExecutingContext to disable the "runIfState()" check
+        MockExecutingContext ctx =
+                new MockExecutingContext() {
+                    @Override
+                    public void runIfState(State expectedState, Runnable action) {
+                        action.run();
+                    }
+                };
```

Comment:
This looks a bit like a hack. Maybe it is because we should have a more targeted test for `StateWithExecutionGraph` where we test when `onGloballyTerminalState` is called.

Reply:
If you got a hammer, everything is a nail. I agree that the tests are misplaced here. I'll remove them from this PR and introduce them through a new ticket: https://issues.apache.org/jira/browse/FLINK-21348